server/src/main/java/com/broadcom/lsp/cobol/core/semantics/SemanticContext.java

```diff
@@ -29,15 +30,18 @@
  * variables, paragraphs, copybooks etc., with their definitions and usages.
  */
 @Value
+@Builder
 @AllArgsConstructor
 public class SemanticContext {
-  private Map<String, Collection<Location>> variableDefinitions;
-  private Map<String, Collection<Location>> variableUsages;
-  private Map<String, Collection<Location>> paragraphDefinitions;
-  private Map<String, Collection<Location>> paragraphUsages;
-  private Map<String, Collection<Location>> constantDefinitions;
-  private Map<String, Collection<Location>> constantUsages;
-  private Map<String, Collection<Location>> copybookDefinitions;
-  private Map<String, Collection<Location>> copybookUsages;
-  private List<DocumentSymbol> outlineTree;
+  Map<String, Collection<Location>> variableDefinitions;
+  Map<String, Collection<Location>> variableUsages;
+  Map<String, Collection<Location>> paragraphDefinitions;
```

Comment:
Why not  List<Location> ?  in UseCasePreprocessorListener, AnalysisResult they are List, so better to keep same

Reply:
It is a result of `Multimap.asMap()`

Reply:
In that case, CobolVisitor becomes more complicated and here we have to convert `Multimap<String, Location>` explicetely:
````
  @NonNull
  public SemanticContext getSemanticContext() {
    return new SemanticContext(
        variables.getDefinitions().asMap(),
        variables.getUsages().asMap(),
        paragraphs.getDefinitions().asMap(),
        paragraphs.getUsages().asMap(),
        sections.getDefinitions().asMap(),
        sections.getUsages().asMap(),
        constants.getDefinitions().asMap(),
        constants.getUsages().asMap(),
        copybooks.getDefinitions().asMap(),
        copybooks.getUsages().asMap(),
        buildOutlineTree());
  }
````

Reply:
It is a known issue, to be fixed after introducing the type check for variables